--- a/gradia/backend/ocr.py	2025-11-18 18:25:10.000000000 +0100
+++ b/gradia/backend/ocr-2.py	2026-01-26 02:21:53.623650067 +0100
@@ -65,7 +65,8 @@
     def __init__(self, window=None):
         self.tesseract_cmd = ocr_tesseract_cmd
         self.original_tessdata_dir = ocr_original_tessdata
-        self.user_tessdata_dir = os.path.expanduser(f"~/.var/app/{app_id}/data/tessdata")
+        xdg_data_home = os.getenv("XDG_DATA_HOME", os.path.expanduser("~/.local/share"))
+        self.user_tessdata_dir = os.path.join(xdg_data_home, app_id, "tessdata")
         self.window = window
 
         pytesseract.pytesseract.tesseract_cmd = self.tesseract_cmd
